shapers/harfbuzz.lua

```diff
@@ -69,13 +66,21 @@ function shaper.getFace (opts)
   SU.debug("fonts", "Resolved font family", opts.family, "->", face and face.filename)
   if not face or not face.filename then SU.error("Couldn't find face '"..opts.family.."'") end
   if SILE.makeDeps then SILE.makeDeps:add(face.filename) end
-  if bitshim.rshift(face.index, 16) ~= 0 then
-    SU.warn("GX feature in '"..opts.family.."' is not supported, fallback to regular font face.")
-    face.index = bitshim.band(face.index, 0xff)
+  face.variations = opts.variations
+  face.pointsize = ("%g"):format(SILE.measurement(opts.size):tonumber())
+
+  -- Try instantiating the font, hb.instanciate() will return nil if it is not
+  -- a variable font or if instanciation failed.
+  face.tempfilename = face.filename
+  local data = hb.instanciate(face)
+  if data then
+    local tmp = os.tmpname();
+    local file = io.open(tmp, "wb")
+    file:write(data)
+    file:close()
+    face.tempfilename = tmp
```

Comment:
We are never delete these temporary files and they litter the tmp dir. They need to be delete after the PDF file is created, but I don’t know if SILE has facilities to run code at such stage. May be lua garbage collection can be utilized, but defining `__gc` on the font table does not seem to ever get called.

Reply:
I think the best analog of what you're doing in the `libtexpdf` source code is `texpdf_close_images` which gets called by `texpdf_close_document`. There's already a `texpdf_close_fonts` but it doesn't delete temporary files like `texpdf_close_images`. `struct pdf_ximage` has a field called `tempfile` which when set causes `dpx_delete_temp_file` to get called on `pdf_ximage.filename`.

Reply:
Thanks for the suggestion, that was helpful. But since the files are not created by libtexpdf, I decided to delete them from the Lua side where they are created.